core/trino-main/src/main/java/io/trino/sql/planner/optimizations/LimitPushDown.java

```diff
@@ -124,7 +124,7 @@ public PlanNode visitLimit(LimitNode node, RewriteContext<LimitContext> context)
                         ImmutableList.of());
             }
 
-            if (!node.isWithTies() || (limit != null && node.getCount() >= limit.getCount())) {
+            if (!node.isOrderSensitive() && (!node.isWithTies() || (limit != null && node.getCount() >= limit.getCount()))) {
```

Comment:
Technically, this should be doable if the source of the limit node's source guarantees the ordering

Reply:
That's right but i'm unsure about the implementation here.
I think doing that would involve tracking ordering scheme in LimitContext while dealing with order sensitivity of each plan node during traveseral.
I'm wondering why this visitor based rule is even needed now that we have Push Limit related iterative optimizer rules ?
Can we skip this one if this rule planned to be eventually deprecated ? LimitPushdown runs much before the TopN to Order sensitive Limit conversion happens anyway.

Reply:
Yes, that's ok. We should follow up on removing the legacy LimitPushDown optimizer. I think all the pieces are in place to be able to do so now.

Reply:
I've filed https://github.com/trinodb/trino/issues/7907 to track removal of legacy LimitPushDown optimizer.